storage/gcs/gcs.go

```diff
@@ -399,6 +399,86 @@ func (s *Storage) Sequence(ctx context.Context, leaf []byte) (uint64, error) {
 	return s.pool.Add(leaf)
 }
 
+// NextAvailable returns the next available unassigned index.
+func (s *Storage) NextAvailable(ctx context.Context) (uint64, error) {
+	tx, err := s.dbPool.BeginTx(ctx, nil)
```

Comment:
I think you don't need a transaction since we're not updating anything here, so this method can be even shorter by just doing `r := s.dbPool.QueryRowContext(...)`

Reply:
That's great, thanks!